src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -3953,4 +3964,116 @@ inline void rpp_store24_f32pkd3_to_f32pkd3_avx(Rpp32f* dstPtr, __m256 *p)
     _mm256_storeu_ps(dstPtr + 16, p[2]); /* Store RGB set 3 */
 }
 
+inline void rpp_convert24_pkd3_to_pln3(__m128i &pxLower, __m128i &pxUpper, __m128i *pxDstChn)
+{
+    // pxLower = R1 G1 B1 R2 G2 B2 R3 G3 B3 R4 G4 B4 R5 G5 B5 R6
+    // pxUpper = G6 B6 R7 G7 B7 R8 G8 B8 0  0  0  0  0  0  0  0
```

Comment:
If you load into pxLower and pxUpper like below, things are a bit more uniform:
```
pxLower = R1G1B1R2G2B2R3G3B3R4G4B4<rest doesn't matter>
pxUpper = R5G5B5R6G6B6R7G7B7R8G8B8<rest doesn't matter>
```
You can then use the already available xmm_char_maskR, xmm_char_maskG, xmm_char_maskB from rpp_cpu_simd.hpp
The xmm_char_maskR will give you R1R2R3R4 from pxLower, and R5R6R7R8 from pxUpper that can be blended.
Similarly two shuffles and a blend for G, and the same for B. 

Would be better from a readability standpoint

Reply:
Okay
Actually the loads for PKD3-PKD3 and PKD3-PLN3 are similar for most of variants
since for PKD3 we need to have the data in continuous manner, we are following the same approach for PKD3-PLN3

if we break this continuity from loads, we need to separate code for PKD3-PLN3 alone for all kernel sizes where this function is used